LoggerManagerのログファイル出力機能が正しく動作していない

# LoggerManagerのログファイル出力機能が正しく動作していない

## 説明
統合テスト実行時に、ロガーがコンソールには出力していますが、ログファイルを正しく作成・書き込みできていません。

## 再現手順
1. pytest tests/integration/test_error_handling_flow.py::TestErrorHandlingFlow::test_logger_outputを実行する
2. 以下のエラーが表示される：
   `
   AssertionError: assert False
   where False = exists()
   `

## 期待される動作
LoggerManagerは環境変数LOG_FILEで指定されたパスにログファイルを作成し、そこにログを書き込むべきです。

## 修正案
core/logger_manager.py内のログファイル処理を修正：

`python
def __init__(self):
    # ...
    log_file = os.environ.get('LOG_FILE')
    if log_file:
        # ディレクトリが存在しない場合は作成
        log_dir = os.path.dirname(log_file)
        if log_dir and not os.path.exists(log_dir):
            os.makedirs(log_dir, exist_ok=True)
            
        # ファイルハンドラを追加
        file_handler = logging.FileHandler(log_file, encoding='utf-8')
        file_handler.setFormatter(formatter)
        logger.addHandler(file_handler)
    # ...
`

## 関連ファイル
- tests/integration/test_error_handling_flow.py
- core/logger_manager.py

--- a/core/logger_manager.py
+++ b/core/logger_manager.py
@@ -5,6 +5,7 @@ import sys
 from pathlib import Path
 from logging.handlers import RotatingFileHandler
 from datetime import datetime
+import os
 
 class LoggerManager:
     """
@@ -21,43 +22,87 @@ class LoggerManager:
             app_name (str): アプリケーション名
         """
         self.app_name = app_name
-        self.log_level = log_level
+        
+        # 環境変数からログレベルを取得
+        log_level_env = os.environ.get('LOG_LEVEL')
+        if log_level_env:
+            if log_level_env == 'DEBUG':
+                self.log_level = logging.DEBUG
+            elif log_level_env == 'INFO':
+                self.log_level = logging.INFO
+            elif log_level_env == 'WARNING':
+                self.log_level = logging.WARNING
+            elif log_level_env == 'ERROR':
+                self.log_level = logging.ERROR
+            elif log_level_env == 'CRITICAL':
+                self.log_level = logging.CRITICAL
+            else:
+                self.log_level = log_level
+        else:
+            self.log_level = log_level
+        
+        # 環境変数からログファイルのパスを取得
+        log_file_env = os.environ.get('LOG_FILE')
         
         # ログディレクトリの設定
-        if log_dir is None:
+        if log_file_env:
+            # 環境変数で指定されたログファイルパスを使用
+            log_file_path = Path(log_file_env)
+            self.log_dir = log_file_path.parent
+            self.log_filename = log_file_path.name
+        elif log_dir is None:
             self.log_dir = Path(__file__).parent.parent / 'logs'
+            self.log_filename = None  # 標準の命名規則を使用
         else:
             self.log_dir = Path(log_dir)
+            self.log_filename = None  # 標準の命名規則を使用
             
         # ログディレクトリが存在しない場合は作成
         self.log_dir.mkdir(parents=True, exist_ok=True)
         
         # ルートロガーの設定
         self.root_logger = logging.getLogger()
-        self.root_logger.setLevel(log_level)
+        self.root_logger.setLevel(self.log_level)
         
         # すでに設定されているハンドラがあれば削除
         for handler in self.root_logger.handlers[::]:
+            handler.close()
             self.root_logger.removeHandler(handler)
             
         # ファイルハンドラの設定
-        self._setup_file_handler()
+        self.file_handler = self._setup_file_handler()
         
         # コンソールハンドラの設定
-        self._setup_console_handler()
+        self.console_handler = self._setup_console_handler()
         
         self.logger = logging.getLogger(app_name)
         self.logger.info(f"{app_name} ロガーの初期化が完了しました")
     
+    def __del__(self):
+        """デストラクタ - ハンドラーリソースの解放"""
+        if hasattr(self, 'file_handler') and self.file_handler:
+            self.file_handler.close()
+        if hasattr(self, 'console_handler') and self.console_handler:
+            self.console_handler.close()
+
     def _setup_file_handler(self):
         """
         ファイルハンドラを設定する
-        """
-        today = datetime.now().strftime("%Y-%m-%d")
-        log_file = self.log_dir / f"{self.app_name}_{today}.log"
         
+        Returns:
+            RotatingFileHandler: 設定されたファイルハンドラー
+        """
+        if self.log_filename:
+            # 環境変数で指定されたファイル名を使用
+            log_file = self.log_dir / self.log_filename
+        else:
+            # 標準の命名規則を使用
+            today = datetime.now().strftime("%Y-%m-%d")
+            log_file = self.log_dir / f"{self.app_name}_{today}.log"
+        
+        # ログファイルのパスを文字列に変換して渡す
         file_handler = RotatingFileHandler(
-            filename=log_file,
+            filename=str(log_file),
             maxBytes=10 * 1024 * 1024,  # 10MB
             backupCount=5,
             encoding='utf-8'
@@ -70,10 +115,14 @@ class LoggerManager:
         file_handler.setLevel(self.log_level)
         
         self.root_logger.addHandler(file_handler)
+        return file_handler
     
     def _setup_console_handler(self):
         """
         コンソールハンドラを設定する
+        
+        Returns:
+            StreamHandler: 設定されたコンソールハンドラー
         """
         console_handler = logging.StreamHandler(sys.stdout)
         
@@ -84,6 +133,7 @@ class LoggerManager:
         console_handler.setLevel(self.log_level)
         
         self.root_logger.addHandler(console_handler)
+        return console_handler
     
     def get_logger(self, name=None):
         """

--- a/tests/integration/test_error_handling_flow.py
+++ b/tests/integration/test_error_handling_flow.py
@@ -174,6 +174,15 @@ class TestErrorHandlingFlow:
                 assert "テスト例外: division by zero" in log_content
                 assert "ZeroDivisionError" in log_content
                 assert "Traceback" in log_content
+            
+            # テスト完了前にハンドラをクリーンアップ
+            if hasattr(logger_manager, 'file_handler') and logger_manager.file_handler:
+                logger_manager.file_handler.close()
+            
+            # すべてのハンドラを明示的に閉じる
+            for handler in logging.getLogger().handlers[:]:
+                handler.close()
+                logging.getLogger().removeHandler(handler)
 
     @patch('services.ebay_scraper.requests.Session.get')
     def test_error_recovery(self, mock_get):

